data/scenarios/Challenges/bucket-brigade.yaml

```diff
@@ -0,0 +1,155 @@
+version: 1
+name: Bucket Brigade
+author: Karl Ostmo
+description: |
+  Convey resources with limited mobility
+creative: false
+robots:
+  - name: base
+    display:
+      char: 'Ω'
+      attr: robot
+    heavy: true
+    dir: [0, 1]
+    devices:
+      - scanner
+      - 3D printer
+      - dictionary
+      - branch predictor
+      - strange loop
+      - clock
+      - string
+      - ADT calculator
+      - comparator
+      - workbench
+      - grabber
+      - lambda
+      - hearing aid
+      - logger
+      - net
+    inventory:
+      - [10, repro kit]
+      - [1, treads]
+  - name: minerobot
+    system: true
+objectives:
+  - goal:
+      - Deliver some "coal" to the base.
+      - |
+        To excavate "coal" from the "mine" (M), a robot needs to
+        `drill` while in posession of a "bucketwheel excavator".
+      - |
+        To assemble the excavator, you'll need to repurpose
+        some "treads".
+        Unfortunately, you have only one set of "treads".
+        You'll have to make do...
+    condition: |
+      try {
+        p <- robotnamed "minerobot";
+        w <- as p {ishere "lignite mine"};
+        return (not w);
+      } { return false }
+  - goal:
+      - Schlep the "coal" back to the base.
+    condition: |
+      as base {has "coal"}
+solution: |
+  run "scenarios/Challenges/_bucket-brigade/brigade.sw"
+entities:
+  - name: bucketwheel excavator
+    display:
+      char: 'x'
+    description:
+      - A powerful machine designed to remove overburden in mining operations.
+    properties: [known, portable]
+    capabilities: [drill]
+  - name: lignite mine
+    display:
+      attr: copper
+      char: 'M'
+    description:
+      - Requires heavy equipment to remove the overburden.
+    properties: [known]
+  - name: coal
+    display:
+      char: 'c'
+    description:
+      - Primitive, sooty fuel.
+    properties: [known, portable]
+  - name: repro kit
+    display:
+      char: 'k'
+    description:
+      - Kit that can be unpacked into everything a robot needs to reproduce.
+    properties: [known, portable]
+recipes:
+  - in:
+      - [1, repro kit]
+    out:
+      - [1, solar panel]
+      - [1, grabber]
+      - [1, 3D printer]
+      - [1, dictionary]
+      - [1, branch predictor]
+      - [1, comparator]
+      - [1, calculator]
+      - [1, lambda]
+      - [1, strange loop]
+      - [1, workbench]
+      - [1, mirror]
+      - [1, string]
+      - [1, clock]
+      - [1, counter]
+      - [1, scanner]
+      - [1, logger]
+      - [1, net]
+      - [1, metal drill]
+      - [1, welder]
+  - in:
+      - [1, treads]
+    out:
+      - [1, bucketwheel excavator]
+  - in:
+      - [1, lignite mine]
+    out:
+      - [1, coal]
```

Comment:
I see that the `lignite mine` is destroyed when you mine it, and thus no additional `coal` is generated.  I was actually kind of disappointed that after I solved the scenario, when I selected "Keep playing" I didn't get to see my coal count continue to go up.  I actually wondered whether I had done something wrong, until I looked at how the scenario is implemented.

Reply:
Looking at other `drill` recipes, I take it that adding `[1, lignite mine]` to the `out` list will prevent it from being destroyed?  I haven't looked into the code regarding that behavior, but there must be special casing that makes `coal` go into your inventory while the `lignite mine` stays on the ground.

Reply:
Yes, in particular, all the `portable` outputs of the recipe go in your inventory, and any other output (of which there must be at most one) stays on the ground.

Reply:
This is also how drilling a `mountain` can turn it into *e.g.* a `mountain tunnel` --- if that recipe is chosen the `mountain` is destroyed but then the non-`portable` `mountain tunnel` is placed in the cell instead.

Reply:
I suppose there could be an interesting mechanic on the `in` side of the recipe as well---perhaps some consumable material like fuel or welding rods could be taken from your inventory as you drill some non-portable object.

Reply:
Yes, that should work, I think.